packages/ag-charts-community/src/util/listeners.ts

```diff
@@ -19,6 +20,12 @@ export class Listeners<EventType extends string, EventHandler extends Handler> {
             this.registeredListeners.set(eventType, [record]);
         }
 
+        for (const pending of this.pendingDispatches) {
+            if (pending.eventType === eventType) {
+                this.dispatch(eventType, ...pending.params);
+            }
+        }
```

Comment:
This feels a little sketchy, as if you have two listeners registering for the events, only one of them will receive the pending events.

Can we sync on Monday about the approach here, I think it might be better to defer some options propagation to the `ChartUpdateType.PERFORM_LAYOUT` rendering lifecycle phase, which might 🤞 mean we don't have to solve race conditions at options injection time?

Reply:
I haven't tested that situation, but it should send it to both listeners. It only clears the pending events when a new one of the same type is dispatched to `dispatchAlwaysResolve()`.